frontend/src/features/public-form/PublicFormProvider.tsx

```diff
@@ -205,6 +206,25 @@ export const PublicFormProvider = ({
         return showErrorToast(error, form)
       }
 
+      const countryRegionFieldIds = new Set(
+        form.form_fields
+          .filter((field) => field.fieldType == BasicField.CountryRegion)
+          .map((field) => field._id),
+      )
+      const formInputsWithCountryRegionInUpperCase = Object.keys(
+        formInputs,
+      ).reduce((newFormInputs: any, fieldId) => {
+        if (
+          countryRegionFieldIds.has(fieldId) &&
+          (formInputs[fieldId] as string)
+        ) {
+          newFormInputs[fieldId] = formInputs[fieldId].toString().toUpperCase()
```

Comment:
maybe declare

```
const currentInput = formInputs[fieldId]
```

and then you can check

```
&& typeof currentInput === 'string' 
```

and you won't need to do `.toString()` before `.toUpperCase()`

Reply:
That's true. Let's go with ^